src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -394,6 +396,10 @@ public SnowflakeStreamingIngestChannelInternal<?> openChannel(OpenChannelRequest
       // Add channel to the channel cache
       this.channelCache.addChannel(channel);
 
+      if (isIcebergMode) {
+        this.flushService.addExternalVolume(request.getFullyQualifiedTableName(), response);
```

Comment:
Have only skimmed FlushService but the changes there look like we need something more intentional:
1. Stage management in a separate class, instead of in FlushService
2. Pass that StageManager into flushService, like the channelCache is passed in
3. IngestClientInternal (this place) directly adds the stage info to the stagemanager, instead of to flushService.

Think of it as - flush service is responsible for flushing to a stage, but not really managing addition / removal of stages, or of renewing the tokens for a stage.

As an aside, this is a good set of concepts esp the S and the O - https://en.wikipedia.org/wiki/SOLID